contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -225,6 +227,10 @@ func NewSchedulerServer() *SchedulerServer {
 		// when kube-proxy is running in a non-root netns (init_net); setting this to a non-zero value will
 		// impact connection tracking for the entire host on which kube-proxy is running. xref (k8s#19182)
 		conntrackMax: 0,
+
+		// cadvisor housekeeping related config
+		housekeepingInterval:       1 * time.Second,
```

Comment:
i don't think there's a way to remove it from `km scheduler --help`. km is
a "hyperkube" binary, meaning that it links together several different
binaries into one. since one of those binaries is the kubelet-executor
(which links against cadvisor) and cadvisor is a bad actor (by defining
global flags), we end up with the cadvisor flags showing up alongside the
scheduler flags

this feels somewhat unsatisfying. need to think about a better way to
handle these flags. having both the new and the old flags show up is very
confusing -- even if it does satisfy the build/verification scripts.

thanks for your patience. will need to think on this.



On Thu, Feb 4, 2016 at 10:13 PM, Lin Sun <notifications@github.com> wrote:

> In contrib/mesos/pkg/scheduler/service/service.go
> <https://github.com/kubernetes/kubernetes/pull/20331#discussion_r51972674>
> :
>
> > @@ -225,6 +227,10 @@ func NewSchedulerServer() *SchedulerServer {
> >  		// when kube-proxy is running in a non-root netns (init_net); setting this to a non-zero value will
> >  		// impact connection tracking for the entire host on which kube-proxy is running. xref (k8s#19182)
> >  		conntrackMax: 0,
> > +
> > +		// cadvisor housekeeping related config
> > +		housekeepingInterval:       1 * time.Second,
>
> I am very puzzled on where the housekeeping_interval=10s is specified... I
> searched the cadvisor code, only see 1s is specified. it seems we should
> remove it from 'km scheduler --help', because we don't support passing it
> via the housekeeping_interval with "_" in it (only dash is supported).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20331/files#r51972674>.
>